source/agora/node/Ledger.d

```diff
@@ -42,11 +44,16 @@ public class Ledger
     /// Temporary storage where transactions are stored until blocks are created.
     private Transaction[] storage;
 
+    /// ledger data dir
+    private string data_dir;
+
     /// Ctor
-    public this ()
+    public this (const string data_dir = ".cache")
     {
         auto block = getGenesisBlock();
         this.addNewBlock(block);
+        if(this.data_dir != data_dir)
```

Comment:
You don't need this check, you are in the constructor, therefore you know for sure `this.data_dir` is not initialized.

Reply:
delete it